File: solution/1915. Number of Wonderful Substrings/1915.cpp
class Solution {
public:
    long long wonderfulSubstrings(string word) {
        long long res = 0;
        vector<int> cnt(1 << 10);
        cnt[0] = 1;
        int state = 0;
        for (char& c : word) {
            state ^= (1 << (c - 'a'));
            res += cnt[state];
            for (int i = 0; i < 10; ++i) {
                res += cnt[state ^ (1 << i)];
            }
            ++cnt[state];
        }
        return res;
    }
};

File: solution/1915. Number of Wonderful Substrings/1915.py
class Solution:
    def wonderfulSubstrings(self, word: str) -> int:
        cnt = [0] * (1 << 10)
        cnt[0] = 1
        res = state = 0
        
        for c in word:
            state ^= (1 << (ord(c) - ord('a')))
            res += cnt[state]
            res += sum(cnt[state ^ (1 << i)] for i in range(10))
            cnt[state] += 1
        return res
    
